cloud/google/serviceaccount.go

```diff
@@ -132,31 +132,24 @@ func (gce *GCEClient) createServiceAccount(serviceAccountPrefix string, roles []
 		return "", "", fmt.Errorf("machine count is zero, cannot create service a/c")
 	}
 
-	// TODO: use real go bindings
-	// Figure out what projects the service account needs permission to.
-	projects, err := gce.getProjects(initialMachines)
+	config, err := gce.clusterproviderconfig(cluster.Spec.ProviderConfig)
 	if err != nil {
 		return "", "", err
 	}
 
-	// The service account needs to be created in a single project, so just
-	// use the first one, but grant permission to all projects in the list.
-	project := projects[0]
 	accountId := serviceAccountPrefix + "-" + util.RandomString(5)
 
-	err = run("gcloud", "--project", project, "iam", "service-accounts", "create", "--display-name="+serviceAccountPrefix+" service account", accountId)
+	err = run("gcloud", "--project", config.Project, "iam", "service-accounts", "create", "--display-name="+serviceAccountPrefix+" service account", accountId)
 	if err != nil {
 		return "", "", fmt.Errorf("couldn't create service account: %v", err)
 	}
 
-	email := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", accountId, project)
+	email := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", accountId, config.Project)
 
-	for _, project := range projects {
-		for _, role := range roles {
-			err = run("gcloud", "projects", "add-iam-policy-binding", project, "--member=serviceAccount:"+email, "--role=roles/"+role)
-			if err != nil {
-				return "", "", fmt.Errorf("couldn't grant permissions to service account: %v", err)
-			}
+	for _, role := range roles {
+		err = run("gcloud", "projects", "add-iam-policy-binding", config.Project, "--member=serviceAccount:"+email, "--role=roles/"+role)
```

Comment:
This section should probably be converted to use the GCP SDK rather than exec the gcloud CLI. I have noticed when I run gcp-deployer locally that there is a lot of time being spent in the section of code currently calling "gcloud projects add-iam-policy-binding". While this will be better now that the ```for _, role := range roles``` is not being duplicated for each machine, it is still could be improved. 

Rather than exec the gcloud CLI it would be better to use the GCP SDK directly for the following reasons:
* Overall speed: it's faster to call the SDK / API inline in this program rather than to fork off gcloud. Even for a single SDK call vs creating a new process.
* By going through the API, I *believe* we can get rid of the multiple calls to add-iam-policy-binding and instead pass in a list of bindings and take care of it all with a single API call. See the API tab at this link for a bit more information: https://cloud.google.com/iam/docs/granting-changing-revoking-access. It looks like it allows for a list of bindings to be passed in.
* Simplified environment: this is a small plus, but without this code we won't need to have gcloud installed.

Reply:
I agree, but I think that should happen in a separate small PR after this. Just because I prefer to keep this PR as targeted as possible and it already has a few different things going on. Does that sound ok to you?

Reply:
Yes, sounds like the best approach. Can even take up the task if you have other pressing items. Let me know.